app/controllers/components_controller.rb

```diff
@@ -2,15 +2,19 @@
 
 class ComponentsController < ApplicationController
   def show
-    @component = Component.get(params[:id])
+    @component = Component.get(params[:id]) || not_found
   end
 
   def index
     @components = Component.all
   end
 
   def preview
-    @component = Component.get(params[:component_id])
+    @component = Component.get(params[:component_id]) || not_found
     render layout: 'preview'
   end
+
+  def not_found
+    render status: 404, text: "404 error"
```

Comment:
@andrewgarner: The issue is, when I do that, I get the following exception when testing non-existent components:

```
Missing template components/not_found, application/not_found with {:locale=>[:en], :formats=>[:html], :variants=>[], :handlers=>[:erb, :builder, :raw, :ruby, :jbuilder]}. Searched in: * "/var/govuk/govuk-component-guide/app/views" * "#<Slimmer::ComponentResolver:0x007fb958c0f5a8>"
```